lib/dialects/abstract/query-generator.js

```diff
@@ -1766,6 +1767,11 @@ module.exports = (function() {
         value = (value.$between || value.$notBetween).map(function (item) {
           return self.escape(item);
         }).join(' AND ');
+      } else if (value && ((value.$like && (any = value.$like.$any) && (comparator = 'LIKE ANY')) ||
```

Comment:
This code is not very readable! It took me time to figure out that the if cases were actually also assignments